tutorials/jul-logs-redirect/src/main/java/TestApp.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.log4j.BasicConfigurator;
+import org.slf4j.LoggerFactory;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+/**
+ * This example demonstrates how to enable JxBrowser logging using SLF4J API. This can be useful
+ * when you are using third party logging libraries such as log4j, logback, etc. Since the JxBrowser
+ * logging system is built on JUL (java.util.logging) you need to route all incoming log records to
+ * the SLF4J API to avoid losing them.
+ *
+ * <p>
+ * Test application represents the simple Swing component for rendering web content.
+ */
+public final class TestApp {
+
+    // The name of the j.u.l root logger that the JxBrowser uses.
+    private static final String ROOT_LOGGER_NAME = "com.teamdev.jxbrowser";
+
+    // Logger you use in your application.
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(TestApp.class);
+
+    public static void main(String[] args) {
+
+        // Configure JxBrowser logging levels.
+        Logger.level(Level.ALL);
+
+        // Configure log4j.
+        BasicConfigurator.configure();
+
+        // Get j.u.l root logger.
+        java.util.logging.Logger julRootLogger = LogManager.getLogManager()
+                .getLogger(ROOT_LOGGER_NAME);
+
+        // Remove existing handlers attached to j.u.l root logger.
+        for (java.util.logging.Handler handler : julRootLogger.getHandlers()) {
+            julRootLogger.removeHandler(handler);
+        }
+
+        // Add SLF4JBridgeHandler to j.u.l's root logger to route all incoming j.u.l records to the SLF4j API.
+        julRootLogger.addHandler(new SLF4JBridgeHandler());
+
+        logger.info("Application started.");
+
+        // Creating and running Chromium engine.
+        Engine engine = Engine.newInstance(HARDWARE_ACCELERATED);
+        Browser browser = engine.newBrowser();
+
+        SwingUtilities.invokeLater(() -> {
+            // Creating Swing component for rendering web content
+            // loaded in the given Browser instance.
+            final BrowserView view = BrowserView.newInstance(browser);
+
+            // Creating and displaying Swing app frame.
+            JFrame frame = new JFrame("Hello World");
+
+            // Close Engine and onClose app window.
+            frame.addWindowListener(new WindowAdapter() {
+                @Override
+                public void windowClosing(WindowEvent e) {
+                    engine.close();
+                    logger.debug("Window closed.");
+                }
+            });
+            frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+            JTextField addressBar = new JTextField("https://google.com");
+            addressBar.addActionListener(e -> browser.navigation()
+                    .loadUrl(addressBar.getText()));
+            frame.add(addressBar, BorderLayout.NORTH);
+            frame.add(view, BorderLayout.CENTER);
+            JButton print = new JButton("Print");
```

Comment:
Please remove all non-essential UI elements from the application. In fact, we don't even need Swing.